.gitlab-ci-check-golang-unittests.yml

```diff
@@ -79,13 +83,16 @@ test:unit:
     - cp -r ${CI_PROJECT_DIR} /go/src/github.com/mendersoftware/${CI_PROJECT_NAME}
     - cd /go/src/github.com/mendersoftware/${CI_PROJECT_NAME}
   script:
-    - go list ./... | grep -v vendor | xargs -n1 -I {} -P 4 go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {} || exit $?
+    - go list ./... | grep -v vendor | xargs -n1 -I {} go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {} 2>&1 | tee /dev/stderr | go-junit-report > ${CI_PROJECT_DIR}/test-results.xml || exit $?
```

Comment:
Because of the extra piping, the `exit` at the end will not detect errors anymore. You need to call `set -o pipefail` before calling this pipeline.

Reply:
Uhm, the test pipeline I launched making the tests fail explicitly seems to work:
https://gitlab.com/Northern.tech/Mender/inventory/-/pipelines/242733895

Are you sure the exit doesn't propagate?

Reply:
I'm sure that the shell doesn't do it by default. But maybe `pipefail` is already on from Gitlab's side? If you've tested it, feel free to disregard this comment.

Reply:
Yep, I tested it because I wanted to be sure the tests that fail will trigger a job failure.